### PR TITLE
Refactor edit profile controller into traits

### PR DESCRIPTION
**Pull request by mariogalic**
## What does this change?

`EditProfileController.scala` has become too large. This PR attempts to break it down into components.

* Move all the Edit Profile functionality into separate `controllers.editprofile` package
* common form handling is in `EditProfileFormHandling` and `ProfileForms`
* Each tab gets its own trait: `AccountTab`, `EmailTab`, `PublicTab`, `SupporterTabs` 
* `ConsentsJourney` gets its own trait
* `EditProfileController` is used only to wire things up

## What is the value of this and can you measure success?

Refactoring.

### CODE REVIEW

File: identity/app/controllers/editprofile/tabs/EmailsTab.scala [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import utils.ConsentOrder.userWithOrderedConsents
+import scala.concurrent.Future
+
+trait EmailsTab
+    extends EditProfileControllerComponents
+    with EditProfileFormHandling {
+
+  import authenticatedActions._
+
+  /** GET /consents/all */
```
Comment (jfsoul) [inline identity/app/controllers/editprofile/tabs/EmailsTab.scala]: Should the consent journey have it's own component outside `EmailsTab`?

File: identity/app/controllers/editprofile/editprofile.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package controllers
+
+import model.IdentityPage
+
+package object editprofile {
```
Comment (jfsoul) [inline identity/app/controllers/editprofile/editprofile.scala]: Fancy

**Review by jfsoul — APPROVED** [review]
Can't argue with these changes, seems more comprehensible now to me.  I would separate consent journey stuff if possible though.

File: identity/app/controllers/editprofile/EditProfileFormHandling.scala [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+            case formData: UserFormData =>
+              identityApiClient.saveUser(userDO.id, formData.toUserUpdateDTO(userDO), userDO.auth) flatMap {
+                case Left(idapiErrors) =>
+                  logger.error(s"Failed to process ${page.id} form submission for user ${userDO.getId}: $idapiErrors")
+                  profileFormsView(page, boundProfileForms.withErrors(idapiErrors), userDO)
+
+                case Right(updatedUser) => profileFormsView(page, boundProfileForms.bindForms(updatedUser), updatedUser)
+              }
+          } // end of success
```
Comment (rahilb) [inline identity/app/controllers/editprofile/EditProfileFormHandling.scala]: these comments are meaningless and can be removed.

File: identity/app/controllers/editprofile/tabs/EmailsTab.scala [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                logger.error(s"Failed to process marketing consent form submission for user ${userDO.getId}: $idapiErrors")
+                InternalServerError(Json.toJson(idapiErrors))
+
+              case Right(updatedUser) =>
+                val successMsg = s"Successfully updated marketing consent for user ${userDO.getId}"
+                logger.info(successMsg)
+                Ok(successMsg)
+            }
+          }
+        ) // end bindFromRequest.fold(
```
Comment (rahilb) [inline identity/app/controllers/editprofile/tabs/EmailsTab.scala]: comments can be removed.

**Review by rahilb — APPROVED** [review]
🚢

**Review by walaura — APPROVED** [review]
LGTM

**Pull request merged**